### PR TITLE
PW-2887: Added TestCard API service

### PR DESCRIPTION
**Pull request by saquibsayyad**
**Description**
Added TestCards API service (/createTestCardRanges)
https://docs.adyen.com/development-resources/test-cards/create-test-cards

**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

File: src/main/java/com/adyen/model/testcard/TestCardRange.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    private AvsAddress address = null;
+
+    @SerializedName("cardHolderName")
+    private String cardHolderName = null;
+
+    @SerializedName("cvc")
+    private String cvc = null;
+
+    @SerializedName("expiryDates")
+    private List<ExpiryDate> expiryDates = null;
```
Comment (Aleffio) [inline src/main/java/com/adyen/model/testcard/TestCardRange.java:39]: the specs show an expiryMonth(enum) and expiryYear, did you generate these models from there?

File: src/main/java/com/adyen/model/testcard/CreateTestCardRangesRequest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+
+public class CreateTestCardRangesRequest {
+    @SerializedName("accountCode")
+    private String accountCode = null;
+
+    @SerializedName("accountTypeCode")
+    private AccountCodeType accountTypeCode = null;
```
Comment (Aleffio) [inline src/main/java/com/adyen/model/testcard/CreateTestCardRangesRequest.java:32]:
```suggestion
    private AccountTypeCode accountTypeCode = null;
```

File: src/main/java/com/adyen/service/resource/testcard/TestCard.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.adyen.Client;
+import com.adyen.Service;
+import com.adyen.service.Resource;
+
+public class TestCard extends Resource {
+
+    public TestCard(Service service) {
+        super(service, service.getClient().getConfig().getEndpoint() + Client.TEST_CARD_API_PAL_SUFFIX + Client.TEST_CARD_API_VERSION + "/createTestCardRanges", null);
+    }
```
[the remark this answers is not in the document]
Comment (saquibsayyad) [reply src/main/java/com/adyen/service/resource/testcard/TestCard.java:31]: I think live as well. In the document there is a section where it is mentioned to switch to live endpoints.

https://docs.adyen.com/development-resources/test-cards/create-test-cards/test-cards-api

File: src/main/java/com/adyen/model/testcard/TestCardRange.java (hunk 1 above)
Comment (saquibsayyad) [reply src/main/java/com/adyen/model/testcard/TestCardRange.java:39]: yes, `ExpiryDate` contains `expiryMonth` enum and `expiryYear`

File: src/main/java/com/adyen/model/testcard/TestCardRange.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    public void setThreeDDirectoryServerResponse(ThreeDSecureData.DirectoryResponseEnum threeDDirectoryServerResponse) {
+        this.threeDDirectoryServerResponse = threeDDirectoryServerResponse;
+    }
+
+    @Override
+    public String toString() {
+        return "TestCardRange{" +
+                "address=" + address +
+                ", cardHolderName='" + cardHolderName + '\'' +
+                ", cvc='***'" +
```
[the remark this answers is not in the document]
Comment (saquibsayyad) [reply src/main/java/com/adyen/model/testcard/TestCardRange.java]: corrected

File: src/test/java/com/adyen/TestCardTest.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Java API Library
+ *
+ * Copyright (c) 2018 Adyen B.V.
```
[the remark this answers is not in the document]
Comment (saquibsayyad) [reply src/test/java/com/adyen/TestCardTest.java]: corrected

File: src/main/java/com/adyen/service/resource/testcard/TestCard.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+package com.adyen.service.resource.testcard;
+
+import com.adyen.Client;
+import com.adyen.Service;
+import com.adyen.service.Resource;
+
+public class TestCard extends Resource {
+
+    public TestCard(Service service) {
+        super(service, service.getClient().getConfig().getEndpoint() + Client.TEST_CARD_API_PAL_SUFFIX + Client.TEST_CARD_API_VERSION + "/createTestCardRanges", null);
```
[the remark this answers is not in the document]
Comment (saquibsayyad) [reply src/main/java/com/adyen/service/resource/testcard/TestCard.java]: corrected

File: src/main/java/com/adyen/service/resource/testcard/TestCard.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ */
+package com.adyen.service.resource.testcard;
+
+import com.adyen.Client;
+import com.adyen.Service;
+import com.adyen.service.Resource;
+
+public class TestCard extends Resource {
```
[the remark this answers is not in the document]
Comment (saquibsayyad) [reply src/main/java/com/adyen/service/resource/testcard/TestCard.java]: corrected

File: src/main/java/com/adyen/model/testcard/CreateTestCardRangesRequest.java (hunk 2 above)
Comment (saquibsayyad) [reply src/main/java/com/adyen/model/testcard/CreateTestCardRangesRequest.java]: corrected

**Pull request closed without merging**